### PR TITLE
fix(username updater): suppress user not found errors in username updater

### LINKED ISSUE #116: Gracefully handle user not found errors in getting response
Gracefully handle the situation where a user has left the Webex Team but exists in the database due to having previously given or received a peg.

Likely in the UpdateUsers method, where the user no longer exist.

Potential resolution: persist the user with their existing name.

### PR DESCRIPTION
**Pull request by Evangelinexx**
so that finish doesn't stop if the usernames can't be updated

fixes #116

### CODE REVIEW

**Review by Chromadream — APPROVED** [review]

**Review by Lauraducky — DISMISSED** [review]

File: src/PockyBot.NET/Services/UsernameUpdater.cs [hunk 1]
```diff
@@ -38,5 +43,18 @@ public async Task<List<PockyUser>> UpdateUsernames(List<PockyUser> users)
...
+        private Task<Person> GetUser(string userId, string username)
+        {
+            try
+            {
+                return _personHandler.GetPersonAsync(userId);
+            }
+            catch (HttpRequestException ex)
+            {
+                _logger.LogError($"Error retrieving details for user {userId}");
+                return Task.FromResult(new Person{UserId = userId, Username = username});
```
Comment (jimcullenaus) [inline src/PockyBot.NET/Services/UsernameUpdater.cs:56]: this isn't going to do anything weird like try to send a PM to a non-existent user, is it?

Comment (Evangelinexx) [reply src/PockyBot.NET/Services/UsernameUpdater.cs:56]: hmmm... worth investigating, good point

**Review by jimcullenaus — CHANGES_REQUESTED** [review]
Requesting changes so we don't accidentally merge this before investigating whether it sends PMs to non-existent users.

**Review by Lauraducky — APPROVED** [review]

**Review by Chromadream — APPROVED** [review]

File: src/PockyBot.NET/PockyBotFactory.cs [hunk 2]
```diff
@@ -28,14 +28,14 @@ public static IPockyBot CreatePockyBot(PockyBotSettings settings, IChatHelper ch
             var pegRequestValidator = new PegRequestValidator(wrappedSettings, configRepository);
             var pegHelper = new PegHelper(configRepository);
             var pegGiver = new PegGiver(pegRepository, chatHelper);
-            var directResultsMessageSender = new DirectResultsMessageSender(chatHelper.Messages);
+            var directResultsMessageSender = new DirectResultsMessageSender(chatHelper.Messages, loggerFactory.CreateLogger<DirectResultsMessageSender>());
```
Comment (jimcullenaus) [inline src/PockyBot.NET/PockyBotFactory.cs:31]: Codecov is complaining about this line. Out of curiosity I wonder if Codecov can be amended to ignore this file.

**Review by jimcullenaus — APPROVED** [review]

**Pull request merged**